### ISSUE TITLE
Geolocation.watchPosition() only fires once even when Location Guard is "paused"

### ISSUE THREAD

**Issue by fabacab** [chatziko/location-guard#14]
I noticed a peculiarity that I think might be a bug but, or it might be desired behavior. Let me know.

When I call [`navigator.geolocation.watchPosition()`](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/watchPosition), I expect that the callback function be called repeatedly. When the Location Guard add-on is enabled, regardless of its settings, the `watchPosition()` callback is only fired once. When I disable the Location Guard add-on, `watchPosition()` behaves as expected.

I have also tried to "Pause" Location Guard but the same issue exists. My expectation was that when Location Guard is "Paused" it would functionally identically to when Location Guard was disabled; this seems to not be the case. Is that a bug?

Here is a test case:

``` html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <title>Location Guard test case</title>
</head>
<body>
    <p>Current position is: <span id="current-position"></span>, updated at <span id="updated-time"></span></p>
    <script>
        navigator.geolocation.watchPosition(updateMyPosition);
        function updateMyPosition (pos) {
            var el = document.getElementById('current-position');
            el.textContent = pos.coords.latitude + ',' + pos.coords.longitude;
            var el = document.getElementById('updated-time');
            el.textContent = new Date().toString();
        }
    </script>
</body>
</html>
```

Loading the above in Firefox 42.0 with Location Guard version 1.2.1 enabled, even if it's paused, the `updated-time` will never change after the first time `updateMyPosition` is called. Disabling Location Guard, however, will show that `updateMyLocation` is called repeatedly as per the browser default.

**Comment by chatziko** [comment]
You're right, this is a bug, thanks for the report.

It's intended that the callback is only called once when location guard is enabled (repeated calls would leak information), but it should be called repeateadly when location guard is paused or set to "use real position". I'll fix it when I find some time.

**Comment by meitar** [comment]
Okay, thanks for confirming this. As this is relevant to a project of mine right now I may be able to find some time to fix it myself. I will post a pull request if I am able to do so.

**Comment by mgarcia4** [comment]
Author, why don't you charge $$ for the app and provide better customer support. I love the app but hate the bugs. I'd be more than willing to sign up for a monthly subscription if your tool was bug free. Think about it. Unfortunately, I haven't found an app that does what yours does.

**Comment by meitar** [comment]
> Author, why don't you charge $$ for the app and provide better customer support. I love the app but hate the bugs. I'd be more than willing to sign up for a monthly subscription if your tool was bug free. Think about it. Unfortunately, I haven't found an app that does what yours does.

As I am not the author of this tool, I realize the above was not directed to me, but as a developer of Free Software who often gets asked the very same question, I feel compelled to provide a brief reply, which is this: in my experience, software one pays money for is not free of bugs.

**Comment by chatziko** [comment]
This pretty old bug is finally fixed in v2.5.0 (to appear online soon).

Now watchPosition works fine in paused / "use real location" mode. Location Guard still shows you (in the icon) the number of times the position is updated, so that the user knows how much information is provided to the website.
